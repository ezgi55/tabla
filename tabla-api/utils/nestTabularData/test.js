const nestTabularData = require(".");
const testData = require("./testResults.json");

it("should nest tabular result set", () => {
  expect(
    nestTabularData(testData, [
      {
        mergeField: "integration_id",
        childrenLabel: "clientIntegrations",
        fieldsToKeep: ["name", "client_id"]
      },
      {
        mergeField: "client_integration_id",
        childrenLabel: "clientSalesChannels",
        fieldsToKeep: ["status", "token"]
      }
    ])
  ).toMatchSnapshot();
});
 