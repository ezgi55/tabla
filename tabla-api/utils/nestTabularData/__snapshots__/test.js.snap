// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should nest tabular result set 1`] = `
Array [
  Object {
    "clientIntegrations": Array [
      Object {
        "clientSalesChannels": Array [
          Object {
            "client_integration_name": "VisioSound Amazon North Americ",
            "client_sales_channel_id": 506,
            "sales_channel_id": 7,
            "sales_channel_ref": "VisioSound Amazon North America Store - Amazon USA",
          },
        ],
        "client_integration_id": "undefined",
        "status": "CONNECTED",
        "token": undefined,
      },
    ],
    "client_id": 1,
    "integration_id": "2",
    "name": "Amazon North America",
  }, 
  Object {
    "clientIntegrations": Array [
      Object {
        "client_integration_name": "Manual Imports - 14580700-9856",
        "client_sales_channel_id": 34,
        "sales_channel_id": 19,
        "sales_channel_ref": "RESEND",
      },
      Object {
        "client_integration_name": "Manual Imports - e081a48d-0a3d",
        "client_sales_channel_id": 35,
        "sales_channel_id": 19,
        "sales_channel_ref": "Manual Orders",
      },
    ],
    "client_id": 1,
    "integration_id": "3",
    "name": "Custom",
  },
  Object {
    "clientIntegrations": Array [
      Object {
        "client_integration_name": "Amazon Europe",
        "client_sales_channel_id": 577,
        "sales_channel_id": 1,
        "sales_channel_ref": "Amazon Europe - Amazon DE",
      },
      Object {
        "client_integration_name": "Amazon Europe",
        "client_sales_channel_id": 578,
        "sales_channel_id": 2,
        "sales_channel_ref": "Amazon Europe - Amazon ES",
      },
      Object {
        "client_integration_name": "Amazon Europe",
        "client_sales_channel_id": 579,
        "sales_channel_id": 3,
        "sales_channel_ref": "Amazon Europe - Amazon FR",
      },
      Object {
        "client_integration_name": "Amazon Europe",
        "client_sales_channel_id": 580,
        "sales_channel_id": 6,
        "sales_channel_ref": "Amazon Europe - Amazon UK",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 886,
        "sales_channel_id": 1,
        "sales_channel_ref": "My Amazon Europe Store - Amazon DE",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 887,
        "sales_channel_id": 2,
        "sales_channel_ref": "My Amazon Europe Store - Amazon ES",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 888,
        "sales_channel_id": 3,
        "sales_channel_ref": "My Amazon Europe Store - Amazon FR",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 889,
        "sales_channel_id": 6,
        "sales_channel_ref": "My Amazon Europe Store - Amazon UK",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 895,
        "sales_channel_id": 1,
        "sales_channel_ref": "My Amazon Europe Store - Amazon DE",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 896,
        "sales_channel_id": 2,
        "sales_channel_ref": "My Amazon Europe Store - Amazon ES",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 897,
        "sales_channel_id": 3,
        "sales_channel_ref": "My Amazon Europe Store - Amazon FR",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store",
        "client_sales_channel_id": 898,
        "sales_channel_id": 6,
        "sales_channel_ref": "My Amazon Europe Store - Amazon UK",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store2",
        "client_sales_channel_id": 935,
        "sales_channel_id": 1,
        "sales_channel_ref": "My Amazon Europe Store2 - Amazon DE",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store2",
        "client_sales_channel_id": 936,
        "sales_channel_id": 2,
        "sales_channel_ref": "My Amazon Europe Store2 - Amazon ES",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store2",
        "client_sales_channel_id": 937,
        "sales_channel_id": 3,
        "sales_channel_ref": "My Amazon Europe Store2 - Amazon FR",
      },
      Object {
        "client_integration_name": "My Amazon Europe Store2",
        "client_sales_channel_id": 938,
        "sales_channel_id": 6,
        "sales_channel_ref": "My Amazon Europe Store2 - Amazon UK",
      },
    ],
    "client_id": 1,
    "integration_id": "6",
    "name": "Amazon Europe",
  },
]
`;
